### PR TITLE
fix: verify check suites before declaring build passed

### PR DESCRIPTION
**Pull request by forketyfork**
## Summary

Fixes false positive build status reporting where stepcat declared builds as passed while GitHub still showed workflows as pending.

## Problem

The `waitForChecksToPass` method only verified **check_runs** via `octokit.checks.listForRef()`. However, GitHub has a separate concept of **check_suites** which can be queued/pending even when individual check_runs appear completed.

For example, the GitHub API might return:
| API | Status |
|-----|--------|
| Check Runs | ✅ completed/success |
| Check Suites | ⏳ pending (Build workflow) |

This caused stepcat to prematurely report "build passed" while the Build workflow was still queued.

## Solution

Added `verifyCheckSuites()` method that:
1. Fetches check suites via `octokit.checks.listSuitesForRef()`
2. Filters to suites matching the target SHA
3. Returns `'pending'` if any suite is queued/in_progress → continues polling
4. Returns `'failed'` if any suite has `failure`/`timed_out`/`cancelled` conclusion
5. Returns `'passed'` when all suites are completed successfully

The method is called after check_runs pass but before returning success.

## Test Plan

- [x] Added test: check runs pass but suites pending → continues polling
- [x] Added test: check suites fail → returns false
- [x] Added test: no check suites exist → edge case passes
- [x] Updated existing tests with `listSuitesForRef` mocks
- [x] All 147 tests pass (72 Vitest + 75 Jest)
- [x] Linting passes

### CODE REVIEW

File: backend/github-checker.ts [hunk 1]
```diff
@@ -270,6 +280,63 @@ export class GitHubChecker {
...
+      );
+      return 'pending';
+    }
+
+    // Check for failed suites
+    const failedSuites = suitesForTarget.filter(
+      suite =>
+        suite.conclusion === 'failure' ||
+        suite.conclusion === 'timed_out' ||
+        suite.conclusion === 'cancelled'
```
[the remark this answers is not in the document]
Comment (forketyfork) [reply backend/github-checker.ts]: Added `action_required` to the list of failed conclusions. Workflows requiring manual intervention or environment approvals will now be treated as failed, preventing false positives.
